homeassistant/components/xiaomi_ble/config_flow.py

```diff
@@ -78,16 +177,22 @@ async def async_step_user(
                 continue
             device = DeviceData()
             if device.supported(discovery_info):
-                self._discovered_devices[address] = (
-                    device.title or device.get_device_name() or discovery_info.name
+                self._discovered_devices[address] = Discovery(
+                    title=device.title
+                    or device.get_device_name()
+                    or discovery_info.name,
```

Comment:
Might be nice to make this a helper function since its used above as well

Reply:
Done in 3bb3d650bd6108f8d1a2d4d801396e8c4d0364a4.